dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java

```diff
@@ -443,4 +445,124 @@ public S getStoreAfter(Block block) {
         return transferInput.analysis.runAnalysisFor(
                 node, before, transferInput, nodeValues, analysisCaches);
     }
+
+    /**
+     * Returns a string representation of this.
+     *
+     * @return a string representation of this
+     */
+    public String repr() {
```

Comment:
This is much more verbose.  It would clutter most output and is useful for debugging.  I have updated the description.

Reply:
It is not the case that nothing useful is output.  `Object.toString` outputs the run-time class name and a unique identifier.  Users may prefer that to dozens of lines of output.  If `toString` is overridden, users don't have a way to get less information.  Creating a different method gives each user the option of how much output to put in logs.

Reply:
Another customary difference between `toString()` and `repr()` is that `toString()` reflects the abstraction or client view, whereas `repr()` shows the implementation or representation.  (Not every implementation of `toString()` and `repr()` follows the convention perfectly.)

Reply:
This is the first class where we add a `repr` method, so I would like to understand what the motivation is and don't understand what you mean with "customary".
I see 483 matches for such a `repr` function on GitHub for Java code. This seems popular in Python code.

In general it seems more useful to output concrete information instead of what `Object.toString` gives. If the behavior of `Object.toString` is really desired in some place, one can simply call `getClass` and `hashCode` instead.

So far in similar classes we have overriden `toString`. What makes these classes special to not do the same?

Reply:
> This is the first class where we add a `repr` method, so I would like to understand what the motivation is 

I added text from this conversation to the developer manual.

> In general it seems more useful to output concrete information instead of what `Object.toString` gives.

It is useful to give clients a choice about what information to output.

> So far in similar classes we have overriden `toString`. What makes these classes special to not do the same?

The difference is not the classes, the difference is the methods.